File: ncs/pyncs/test/core/mailertest.py
#!/usr/bin/env python
"""
mailer test suite.
"""

import sys, os
import unittest

# adding current directory to path to make sure local modules can be imported
sys.path.insert(0, '.')
from ncs.core.mailer import Mailer

class MailerTest(unittest.TestCase):
	def setUp(self):
		self.mailer = Mailer('de01exm68.ds.mot.com')
		self.mailer.fromAddr('hzcosim@wimax-cosim.mot.com')
		self.mailer.toAddr(['cwnj74@motorola.com'])

	def test_send(self):
		self.mailer.subject('test python smtp lib')
		self.mailer.content('test content')
		self.mailer.attach(os.getcwd()+"/test/mail_merged.html")
		self.mailer.attach(os.getcwd()+"/test/Coverage_full.csv")
		self.mailer.send()
        print("mail has been sent out!")
		
if __name__ == "__main__":
    unittest.main()
    

File: ncs/rncslib/test/MailerTest.rb
#!/usr/bin/ruby -w

require 'test/unit'
require 'logger'
require 'ncs/Mailer'

class MailerTest < Test::Unit::TestCase
	@@ClassName = 'MailerTest'
	
	def setup
		@mailer = Mailer.new
		@mailer.mailserver = 'de01exm68.ds.mot.com'
		@mailer.from = 'hzcosim@wimax-cosim.mot.com'
		@mailer.tolist = ['cwnj74@motorola.com']
	end
	
	def test_sendhtml
		@mailer.subject = 'test ruby smtp lib'
		@mailer.body = [
			'<div style="color:red;">', 
			'test ruby smtp<br/>', 
			'test ruby sendmail',
			'</div>']
		@mailer.sendhtml()
	end
	
	def test_sendtext
		@mailer.subject = 'test ruby smtp lib'
		@mailer.body = [
			'test ruby smtp', 
			'test ruby sendmail',
			]
		@mailer.sendtext()
	end
	
	def teardown
		@mailer = nil
	end
end